### PR TITLE
Infer resource for plan change

### PR DESCRIPTION
**Pull request by bjeanes**
One option for addressing #28

cc @ddollar @heroku/add-ons

### CODE REVIEW

File: lib/ext/heroku/command/addons.rb [hunk 1]
```diff
@@ -447,6 +406,79 @@ def open
...
+      end
+
+      case addons.count
+      when 1
+        return addons[0]
+      when 0
+        error("No #{service_name} add-on on app #{app_name} found")
+      else
+        error("Ambiguous add-on identifier #{service_plan_specifier}\nList your add-ons with `heroku addons`")
+      end
```
Comment (bjeanes) [inline lib/ext/heroku/command/addons.rb]: More testing is needed for the failure cases here before a merge. It's getting late here so I've put this up for discussion here and in #28

File: lib/ext/heroku/command/addons.rb [hunk 2]
```diff
@@ -447,6 +406,79 @@ def open
...
+    def resolve_addon(app_name, service_plan_specifier)
+      service_name, plan_name = service_plan_specifier.split(':')
+
+      # TODO: temporary until heroku/api#3725 is merged
+      service_id = api.request(
+        :expects  => 200,
+        :headers  => { "Accept" => "application/vnd.heroku+json; version=3" },
+        :method   => :get,
+        :path     => "/addon-services/#{service_name}"
+      ).body["id"]
```
Comment (bjeanes) [inline lib/ext/heroku/command/addons.rb]: This is temporary until heroku/api#3725 is merged, which fixes an issue with the `addon['addon_service']['name']` value (see L464).

**Comment by mnoble** [comment]
I'm cool with this direction. In fact, I should probably update #37 to use `resolve_addon`, since I have similar logic there.

File: lib/ext/heroku/command/addons.rb (hunk 2 above)
Comment (mattreduce) [inline lib/ext/heroku/command/addons.rb]: heroku/api#3725 was merged :boom:

I can't believe I missed this pull-request two weeks ago O.O Thanks, @bjeanes!

**Comment by bjeanes** [comment]
Pulling this in...

**Pull request merged**